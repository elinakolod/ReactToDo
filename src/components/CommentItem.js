import React, { Component } from 'react';
import {connect} from 'react-redux';
import { ListGroup } from 'react-bootstrap';
import { removeCommentRequest } from '../redux/actions';

class CommentItem extends Component {
  removeButtonClick = event => {
    event.preventDefault()
    this.props.removeCommentRequest(this.props.comment.id, this.props.comment.task_id)
  }

  render() {
    return (
      <ListGroup.Item variant="light">
        { this.props.comment.body }
        <a href="#" className="close-comment" onClick={this.removeButtonClick} />
      </ListGroup.Item>
    );
  }
}

const mapDispatchToProps = {
  removeCommentRequest: removeCommentRequest
}

export default connect(null, mapDispatchToProps)(CommentItem);
