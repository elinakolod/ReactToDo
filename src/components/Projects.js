import React, { Component } from 'react';
import {connect} from 'react-redux';
import {fetchProjectsRequest} from '../redux/actions';

class Projects extends Component {
  componentDidMount = () => {
    this.props.fetchProjectsRequest()
  }

  render() {
    return (
      <div>
        this.props.projects
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    projects: state.projects
  }
}

const mapDispatchToProps = dispatch => ({
  fetchProjectsRequest: () => dispatch(fetchProjectsRequest())
})

export default connect(mapStateToProps, mapDispatchToProps)(Projects);
