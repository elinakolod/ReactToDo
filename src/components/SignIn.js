//import axios from 'axios';

//class SignIn extends Component {
  // constructor(props) {
  //   super(props)
  //   this.state = {
  //     headers: []
  //   }
  // }

  // getTasks() {
  //   let params = {
  //     email: 'test10@i.ua',
  //     password: 'BoraBora1',
  //     password_confirmation: 'BoraBora1'
  //   }

  //   axios.post('/auth', params)
  //   .then(response => {
  //     this.setState({headers: response.headers})
  //   })
  //   .catch(error => console.log(error))
  // }

  // componentDidMount() {
  //   this.getTasks()
  // }

  import React, {Component} from 'react';
  import {connect} from 'react-redux';
  import {signInUserRequest} from '../redux/actions';

  class SignIn extends Component {
    state = {
      email: "",
      password: ""
    }

    handleChange = event => {
      this.setState({
        [event.target.name]: event.target.value
      });
    }

    handleSubmit = event => {
      event.preventDefault()
      this.props.signInUserRequest(this.state)
    }

    render() {
      return (
        <form onSubmit={this.handleSubmit}>
          <h1>Login</h1>

          <label>Email</label>
          <input
            name='email'
            placeholder='email'
            value={this.state.email}
            onChange={this.handleChange}
            /><br/>

          <label>Password</label>
          <input
            type='password'
            name='password'
            placeholder='Password'
            value={this.state.password}
            onChange={this.handleChange}
            /><br/>

          <input type='submit'/>
        </form>
      )
    }
  }

  const mapDispatchToProps = {
    signInUserRequest: signInUserRequest
  }

  export default connect(null, mapDispatchToProps)(SignIn);
