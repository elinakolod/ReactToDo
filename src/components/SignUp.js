import React, {Component} from 'react';
import {connect} from 'react-redux';
import {signUpUserRequest} from '../redux/actions';

class Signup extends Component {
  state = {
    firstName: "",
    lastName: "",
    email: "",
    password: ""
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault()
    this.props.signUpUserRequest(this.state)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h1>Sign Up For An Account</h1>

        <label>First Name</label>
        <input
          name='firstName'
          placeholder='first name'
          value={this.state.firstName}
          onChange={this.handleChange}
          /><br/>

        <label>Last Name</label>
        <input
          name='lastName'
          placeholder='last name'
          value={this.state.lastName}
          onChange={this.handleChange}
          /><br/>

        <label>Email</label>
        <textarea
          name='email'
          placeholder='email'
          value={this.state.email}
          onChange={this.handleChange}
          /><br/>

        <label>Password</label>
        <input
          type='password'
          name='password'
          placeholder='Password'
          value={this.state.password}
          onChange={this.handleChange}
          /><br/>

        <input type='submit'/>
      </form>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  signUpUserRequest: userInfo => dispatch(signUpUserRequest(userInfo))
})

export default connect(null, mapDispatchToProps)(Signup);
