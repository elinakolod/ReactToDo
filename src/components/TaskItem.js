import React, { Component } from 'react';
import {connect} from 'react-redux';
import { pick, values, map } from 'lodash';
import CommentItem from './CommentItem';
import { removeTaskRequest, createCommentRequest, updateTaskRequest } from '../redux/actions';
import { Form, Button, InputGroup, FormControl, ListGroup } from 'react-bootstrap';

class TaskItem extends Component {
  state = {
    body: '',
    show: true
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault()
    this.props.createCommentRequest(this.state, this.props.task.id)
  }

  renderComment(comment) {
    return (
      <CommentItem comment={comment} key={comment.id}/>
    )
  }

  removeButtonClick = event => {
    this.props.removeTaskRequest(this.props.task.project_id, this.props.task.id)
  }

  checkTask = event => {
    let task = { done: event.target.checked }
    this.props.updateTaskRequest(task, this.props.task.id)
  }

  editButtonClick = event => {
    let task_id = this.props.task.id
    let input = document.getElementById('taskNameInput' + task_id)
    let name_span = document.getElementById('taskName' + task_id)
    document.getElementById('taskHeader' + task_id).style.display = 'none'
    input.style.display = 'inline'
    input.value = this.props.task.name
    name_span.append(input)
  }

  updateTaskName = event => {
    if (event.key === 'Enter') {
      let task_id = this.props.task.id
      let task = { name: event.target.value }
      this.props.updateTaskRequest(task, task_id)
      event.target.style.display = 'none'
      document.getElementById('taskHeader' + task_id).style.display = 'inline'
    }
  }

  render() {
    return (
      <ListGroup.Item action>
        <input
          type="checkbox"
          defaultChecked={this.props.task.done}
          onChange={this.checkTask} />
        <span id={ 'taskName' + this.props.task.id }>
          <h6
            id={ 'taskHeader' + this.props.task.id }
            className={ this.props.task.done ? "done" : '' }>
            { this.props.task.name }
          </h6>
        </span>
        <input
          id={ 'taskNameInput' + this.props.task.id }
          className= 'taskNameInput'
          onKeyDown={this.updateTaskName} />
        <a href="#" onClick={this.editButtonClick}>Edit</a>
        <a href="#" className="close-task" onClick={this.removeButtonClick} />

        <ListGroup>
          { this.props.comments.map(this.renderComment) }
        </ListGroup>

        <Form onSubmit={this.handleSubmit}>
            <InputGroup className="mb-3">
              <FormControl
                name='body'
                placeholder="Comment's name"
                aria-label="Comment's name"
                aria-describedby="basic-addon2"
                value={this.state.body}
                onChange={this.handleChange}
              />
              <InputGroup.Append>
                <Button variant="primary" type="submit">Add Comment</Button>
              </InputGroup.Append>
            </InputGroup>
          </Form>
      </ListGroup.Item>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    comments: values(pick(state.entities.comments, props.task.comments.map(comment => comment.id)))
  }
}

const mapDispatchToProps = {
  removeTaskRequest: removeTaskRequest,
  createCommentRequest: createCommentRequest,
  updateTaskRequest: updateTaskRequest
}

export default connect(mapStateToProps, mapDispatchToProps)(TaskItem);
