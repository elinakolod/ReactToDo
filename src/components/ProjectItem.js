import React, { Component } from 'react';
import {connect} from 'react-redux';
import { pick, values } from 'lodash';
import TaskItem from './TaskItem';
import { createTaskRequest, removeProjectRequest, updateProjectRequest } from '../redux/actions';
import { Form, Button, InputGroup, FormControl, ListGroup } from 'react-bootstrap';

class ProjectItem extends Component {
  state = {
    new_task_name: '',
    name: this.props.project.name,
    editFormVisible: false
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault()
    this.props.createTaskRequest({ name: this.state.new_task_name }, this.props.project.id)
    this.setState({new_task_name: ''})
  }

  removeButtonClick = event => {
    event.preventDefault()
    this.props.removeProjectRequest(this.props.project.id)
  }

  editButtonClick = event => {
    this.setState({editFormVisible: true})
  }

  updateProjectName = event => {
    if (event.key === 'Enter') {
      let project_id = this.props.project.id
      let project = { name: event.target.value }
      this.props.updateProjectRequest(project, project_id)
      this.setState({editFormVisible: false})
    }
  }

  renderTask(task) {
    return (
      <TaskItem task={task} key={task.id}/>
    )
  }

  render() {
    return (
      <div>
        <span id={ 'projectName' + this.props.project.id }>
          { !this.state.editFormVisible ?
            <h3 id={'projectHeader' + this.props.project.id }>
              { this.props.project.name }
            </h3> : <Form.Control
                      id={ 'projectNameInput' + this.props.project.id}
                      name='name'
                      value={ this.state.name }
                      onKeyDown={this.updateProjectName}
                      onChange={this.handleChange} />
          }
        </span>
        <a href="#" onClick={this.editButtonClick}>Edit</a>
        <Button variant="outline-danger" onClick={this.removeButtonClick}>Remove</Button>
        <div style={{paddingTop: '10px'}}>
          <span class='error'>{ this.props.errors }</span>
          <Form onSubmit={this.handleSubmit}>
            <InputGroup className="mb-3">
              <FormControl
                name='new_task_name'
                placeholder="Task's name"
                aria-label="Task's name"
                aria-describedby="basic-addon2"
                value={this.state.new_task_name}
                onChange={this.handleChange}
              />
              <InputGroup.Append>
                <Button variant="primary" type="submit">Add Task</Button>
              </InputGroup.Append>
            </InputGroup>
          </Form>

          <ListGroup>
            { this.props.tasks.map(this.renderTask) }
          </ListGroup>
        </div>
        <br/>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    tasks: values(pick(state.entities.tasks, props.project.tasks.map(task => task.id))),
    errors: state.entities.errors?.name
  }
}

const mapDispatchToProps = {
  createTaskRequest: createTaskRequest,
  removeProjectRequest: removeProjectRequest,
  updateProjectRequest: updateProjectRequest
}

export default connect(mapStateToProps, mapDispatchToProps)(ProjectItem);
