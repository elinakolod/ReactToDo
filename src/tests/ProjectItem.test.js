import React from 'react';
import {ProjectItem} from '../components/ProjectItem';
import {configure, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux';

const middlewares = [thunk];
const mockStore = configureStore(middlewares)
configure({ adapter: new Adapter() })

describe('ProjectItem', () => {

  const state = {
    entities: {
      projects: {
        0: {id: 1, name: 'Project name', tasks: [{ id: 2, type: 'task' }]}
      },
      tasks: {
        0: {id: 2, name: 'Task name', project_id: 1, comments: [{ id: 3, type: 'comment' }]}
      },
      comments: {
        0: {id: 3, body: 'Comment body', task_id: 1}
      }
    }
  }
  const store = mockStore(state)

  let project = {
    id: 1,
    name: 'Project name'
  }

  const defaultProps = {
    project: project
  }

  it('matches ProjectItem snapshot', () => {
    const wrapper = shallow(<Provider store={store}><ProjectItem {...defaultProps}/></Provider>)

    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
