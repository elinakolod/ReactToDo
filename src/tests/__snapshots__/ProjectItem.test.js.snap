// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem matches ProjectItem snapshot 1`] = `
<div>
  <span
    id="projectName1"
  >
    <h3
      id="projectHeader1"
    />
  </span>
  <a
    href="#"
    onClick={[Function]}
  >
    Edit
  </a>
  <button
    className="btn btn-outline-danger"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
  <div
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="mb-3 input-group"
      >
        <input
          aria-describedby="basic-addon2"
          aria-label="Task's name"
          className="form-control"
          name="new_task_name"
          onChange={[Function]}
          placeholder="Task's name"
          value=""
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Add Task
          </button>
        </div>
      </div>
    </form>
    <div
      className="list-group"
      onKeyDown={[Function]}
    />
  </div>
  <br />
</div>
`;
